File: API/DTOs/Pagination/PaginatedResult.cs
using System.Collections.Generic;

namespace API.DTOs.Pagination
{
    public class PaginatedResult<T> where T : class
    {
        public PaginatedResult(IReadOnlyList<T> items, int totalItemsCount, int pageNumber, int pageSize, int totalPages)
        {
            Items = items;
            TotalItemsCount = totalItemsCount;
            PageNumber = pageNumber;
            PageSize = pageSize;
            TotalPages = totalPages;
        }

        public IReadOnlyList<T> Items { get; set; }
        public int TotalItemsCount { get; set; }
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public int TotalPages { get; set; }
    }
}

File: ClientApp/src/app/shared/models/pagination/paginatedResult.ts
export class PaginatedResult<T> {
    items: T[];
    totalItemsCount: number;
    pageNumber: number;
    pageSize: number;
    totalPages: number;

    constructor(items: T[], totalItemsCount: number, pageNumber: number, pageSize: number, totalPages: number) {
        this.totalItemsCount = totalItemsCount;
        this.pageNumber = pageNumber;
        this.pageSize = pageSize;
        this.totalPages = totalPages;
        this.items = items;
    }
}